src/ui/screens/Parachains/OverviewScreen.tsx

```diff
@@ -121,55 +105,39 @@ const Overview = React.memo(function Overview({
         );
       }}
       data={parachains}
-      keyExtractor={(item) => item.toString()}
-      renderItem={({item}) => <Parachain id={item} leasePeriod={leasePeriod} />}
+      keyExtractor={(item) => item.id.toString()}
+      renderItem={(item) => <ParachainItem item={item.item} key={item.item.id.toString()} />}
     />
   );
 });
 
-function Parachain({id, leasePeriod}: {id: ParaId; leasePeriod?: LeasePeriod}) {
+function ParachainItem({item}: {item: Parachain}) {
   const {navigate} = useNavigation();
-  const {data: leases} = useParachainLeases(id);
-  const endpoints = useParaEndpoints(id);
-  const endpoint = endpoints ? endpoints[endpoints.length - 1] : null;
-
-  const period = useMemo(
-    () => leasePeriod?.currentPeriod && leases && getLeasePeriodString(leasePeriod.currentPeriod, leases),
-    [leasePeriod, leases],
-  );
-
-  const lastLease = leases ? leases[leases.length - 1] : null;
-  const leaseValue = lastLease ? lastLease + 1 : null;
-  const blocks = useMemo(
-    () =>
-      leasePeriod && leaseValue && bnToBn(leaseValue).sub(BN_ONE).imul(leasePeriod.length).iadd(leasePeriod.remainder),
-    [leasePeriod, leaseValue],
-  );
-
-  const parachainName = endpoint?.text?.toString() ?? `#${id.toString()}`;
-
+  const {lease} = item;
   return (
     <>
       <List.Item
+        title={item.name}
+        key={item.id.toString()}
         onPress={() => {
           navigate(parachainDetailScreen, {
-            id: id.toString(),
-            name: parachainName,
-            period,
-            blocks: bnToHex(blocks),
+            parachainId: item.id,
           });
         }}
-        title={parachainName}
         left={() => (
           <View style={globalStyles.justifyCenter}>
-            <Caption>{formatNumber(id.toNumber())}</Caption>
+            <Caption>{item.id}</Caption>
           </View>
         )}
         right={() => {
           return (
             <View style={styles.rightItem}>
-              <Text>{period}</Text>
-              {blocks ? <BlockTime blockNumber={blocks} /> : null}
+              <Text>{lease?.period}</Text>
+              <Text style={globalStyles.rowContainer}>
+                {lease?.blockTime
+                  ? lease.blockTime.slice(0, 2).map((block: string, i: number) => <Text key={i}>{block} </Text>)
```

Comment:
There is no need to add nested `Text` components here, you can use a variable to put the time and pass it here.